### PR TITLE
HIPP-1119: Update a simple API

### PR DESCRIPTION
**Pull request by PaulCDurham**
https://jira.tools.tax.service.gov.uk/browse/HIPP-1119

### CODE REVIEW

File: app/uk/gov/hmrc/apihubapplications/connectors/APIMConnectorImpl.scala [hunk 1]
```diff
@@ -81,39 +79,79 @@ class APIMConnectorImpl @Inject()(
...
+            Left(raiseApimException.unexpectedResponse(response.status.intValue))
+          }
+      )
+  }
+
+  private def handleSuccessfulRequest(response: HttpResponse): Either[ApimException, SuccessfulDeploymentsResponse] = {
+    response.json.validate[SuccessfulDeploymentsResponse].fold(
+      (errors: collection.Seq[(JsPath, collection.Seq[JsonValidationError])]) => Left(raiseApimException.invalidResponse(errors)),
+      deploymentsResponse => {
+        logger.info(s"APIM Response: ${Json.prettyPrint(Json.toJson(deploymentsResponse))}")
```
Comment (rdawson-hmrc) [inline app/uk/gov/hmrc/apihubapplications/connectors/APIMConnectorImpl.scala:130]: Will this add line-breaks into the log message? If so will that harm parseability so any tools expecting single line messages

File: it/test/uk/gov/hmrc/apihubapplications/APIMConnectorSpec.scala [hunk 2]
```diff
@@ -245,95 +251,228 @@ class APIMConnectorSpec
...
+          )
+      )
+
+      buildConnector().redeployToSecondary(publisherRef, redeploymentRequest)(HeaderCarrier()).map {
+        actual =>
+          actual mustBe Left(ApimException.unexpectedResponse(500))
       }
+    }
 
+    "must return ServiceNotFound when the Simple API Deployment service returns 4040 Not Found" in {
```
Comment (rdawson-hmrc) [inline it/test/uk/gov/hmrc/apihubapplications/APIMConnectorSpec.scala:371]: typo - 4040

File: test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala [hunk 3]
```diff
@@ -195,10 +196,132 @@ class DeploymentsControllerSpec
...
+        when(fixture.deploymentsService.redeployToSecondary(any(), any())(any()))
+          .thenReturn(Future.successful(Left(ApimException.unexpectedResponse(500))))
+
+        val result = route(fixture.application, request).value
+        val e = the [ApimException] thrownBy status(result)
+        e mustBe ApimException.unexpectedResponse(500)
+      }
+    }
+
+    "must return 4040 Not Found when the publisher ref does not identify a service known to APIM" in {
```
Comment (rdawson-hmrc) [inline test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala:306]: 4040

File: it/test/uk/gov/hmrc/apihubapplications/APIMConnectorSpec.scala (hunk 2 above)
Comment (PaulCDurham) [reply it/test/uk/gov/hmrc/apihubapplications/APIMConnectorSpec.scala]: Fixed

File: test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala (hunk 3 above)
Comment (PaulCDurham) [reply test/uk/gov/hmrc/apihubapplications/controllers/DeploymentsControllerSpec.scala]: Fixed

**Review by rdawson-hmrc — APPROVED** [review]

**Pull request merged**